### PR TITLE
gendef: init at 11.0.1

### PR DESCRIPTION
**Pull request by hughobrien**
###### Description of changes

> [gendef](https://sourceforge.net/p/mingw-w64/wiki2/gendef/)  is a tool which generate def files from DLLs. A def file is a list of symbols exported by a DLL. The primary use of this tool is to allow creation of a import library of DLLs created by non-GCC compilers. It can handle both x86 (win32) and amd64 (win64) executables.

Gendef is part of mingw-W64

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/gendef/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+  sourceRoot = "mingw-w64/mingw-w64-tools/gendef";
+
+  meta = with lib; {
+    description = "gendef tool from mingw-w64";
+    homepage = "https://sourceforge.net/p/mingw-w64/wiki2/gendef/";
+    license = licenses.gpl3Plus;
+    maintainers = with maintainers; [ hughobrien ];
+    platforms = platforms.linux;
+  };
```
Comment (AndersonTorres) [inline pkgs/development/tools/gendef/default.nix:21]:
```suggestion
  meta = {
    description = "A tool which generate def files from DLLs";
    homepage = "https://sourceforge.net/p/mingw-w64/wiki2/gendef/";
    license = lib.licenses.gpl3Plus;
    maintainers = with lib.maintainers; [ hughobrien ];
    platforms = lib.platforms.linux;
  };
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by hughobrien** [comment]
squashed fixup commit

**Review by drupol — COMMENTED** [review]
Hi!

Thanks for your contribution.

I added a non-blocking comment, feel free to implement it if you want, if not, just let me know and I'll merge this immediately.

File: pkgs/development/tools/gendef/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{ fetchgit, lib, stdenv }:
+
+stdenv.mkDerivation rec {
```
[the remark this answers is not in the document]
Comment (hughobrien) [reply pkgs/development/tools/gendef/default.nix]: Thank you for teaching me about this pattern, I have updated the PR to include it.

**Review by drupol — APPROVED** [review]

**Review by drupol — APPROVED** [review]

**Pull request merged**